hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -974,4 +1159,20 @@ public void close() {
     // Stop timeline-server if running
     super.close();
   }
+
+  protected void updateTableMetadata(HoodieTable table, HoodieCommitMetadata commitMetadata,
+                                   HoodieInstant hoodieInstant,
+                                   HoodieData<WriteStatus> writeStatuses) {
+    // Do not do any conflict resolution here as we do with regular writes. We take the lock here to ensure all writes to metadata table happens within a
+    // single lock (single writer). Because more than one write to metadata table will result in conflicts since all of them updates the same partition.
+    table.getMetadataWriter(hoodieInstant.getTimestamp())
+        .ifPresent(writer -> ((HoodieTableMetadataWriter) writer).update(commitMetadata, writeStatuses, hoodieInstant.getTimestamp()));
+  }
+
+  protected void handleWriteErrors(List<HoodieWriteStat> writeStats, TableServiceType tableServiceType) {
```

Comment:
this was never invoked in FlinkTableServiceClient. 
lets try to keep the functionality as is. 
atleast lets override the method and make it no-op. lets not change any functionality un-intentionally

Reply:
Ok sounds good